src/v/transform/logging/rpc_client.cc

```diff
@@ -0,0 +1,114 @@
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "rpc_client.h"
+
+#include "cluster/topic_table.h"
+#include "hashing/murmur.h"
+#include "model/namespace.h"
+#include "transform/logging/logger.h"
+#include "transform/logging/record_batcher.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/coroutine/as_future.hh>
+
+namespace transform::logging {
+
+rpc_client::rpc_client(
+  rpc::client* rpc_client,
+  cluster::topic_table* topic_table,
+  cluster::metadata_cache* metadata_cache)
+  : _rpc_client(rpc_client)
+  , _topic_table(topic_table)
+  , _metadata_cache(metadata_cache) {}
+
+ss::future<errc>
+rpc_client::write(model::partition_id pid, io::json_batches batches) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    auto cfg = _metadata_cache->get_topic_cfg(ns_tp);
+    if (!cfg.has_value()) {
+        co_return errc::topic_not_found;
+    }
+
+    auto max_batch_size = cfg->properties.batch_max_bytes.value_or(
+      config::shard_local_cfg().kafka_batch_max_bytes());
+
+    record_batcher batcher{max_batch_size};
+
+    while (!batches.empty()) {
+        auto json_batch = std::move(batches.front());
+        batches.pop_front();
+        while (!json_batch.events.empty()) {
+            auto ev = std::move(json_batch.events.front());
+            json_batch.events.pop_front();
+            batcher.append(json_batch.name.copy(), std::move(ev));
+        }
+    }
+
+    auto total_request_size = batcher.total_size_bytes();
+    auto record_batches = batcher.finish();
+
+    model::topic_partition tp{model::transform_log_internal_topic, pid};
+    vlog(
+      tlg_log.debug,
+      "Producing logs for {{ {} }} total size: {}B",
+      tp,
+      total_request_size);
+
+    auto res = co_await _rpc_client->produce(
+      std::move(tp), std::move(record_batches));
+    if (res != cluster::errc::success) {
+        co_return errc::write_failure;
+    }
+    co_return errc::success;
+}
+
+result<model::partition_id, errc>
+rpc_client::compute_output_partition(model::transform_name_view name) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    const auto& config = _topic_table->get_topic_cfg(ns_tp);
+    if (!config) {
+        return errc::topic_not_found;
+    }
+
+    // TODO(oren): Does "is_user_topic" make this unnecessary?
+    const auto* disabled_set = _topic_table->get_topic_disabled_set(ns_tp);
+    if (!(disabled_set && disabled_set->is_fully_disabled())) {
+        auto n_partitions = static_cast<uint32_t>(config->partition_count);
+        for (auto i : boost::irange(n_partitions)) {
+            model::partition_id candidate{static_cast<int32_t>(
+              (murmur2(name().data(), name().size()) + i) % n_partitions)};
+            if (!(disabled_set && disabled_set->is_disabled(candidate))) {
+                return candidate;
+            }
+        }
+    }
```

Comment:
Let's just remove this TBH, since this is transient state, it's better to drop them and attempt to produce to a disabled partition. If a single partition is disabled, then we can just drop those logs during that time, as the readers of this topic probably won't have this information to read from the correct topic and get this information (I assume most code will filter for a single transform by filtering for a single partition).

Reply:
>drop them and attempt to produce to a disabled partition

You're suggesting to optimistically return the calculated partition ID irrespective of the disabled_set and allow the produce request to fail, right?

Reply:
Yupp

Reply:
done